x-pack/legacy/plugins/epm/common/types.ts

```diff
@@ -128,10 +128,15 @@ export interface Dataset {
   release: string;
   ingest_pipeline: string;
   vars?: VarsEntry[];
-  type: string;
+  type: DatasetType;
```

Comment:
shouldn't we make sure to know what types we are dealing with?  and then update it when we support new types?

Reply:
> shouldn't we make sure to know what types we are dealing with?

Yes, and in this case it's any `string`. As soon as a package is added with a different `type` this will be incorrect. We can't count on being made aware of the new package/type and that process gets more important/difficult as the EPR grows.

> and then update it when we support new types?

As far as I can tell, the only place that's currently concerned with `logs` or `metrics` is `installIndexPatterns` and even that calls the same function just with different values. I think we should either delete `DatasetType` or move it to the one file where it's used and lift it back up when/if other code needs it.

Reply:
There's also the practical matter that the `Dataset` type will soon be coming directly from EPR and `type` will only be `string`.

See the [source go struct](https://github.com/elastic/package-registry/blob/3114374a0afb8be92098522c8ffc05d2c7aa62a3/util/dataset.go#L17) and the [resulting TS type](https://github.com/jfsiii/package-registry/commit/71895694ff84dd56aba9a9055c02262dc3171935#diff-d96ec7ae5bb4a0907779f25ed03acb33R26)

Reply:
I guess the question is whether I should be able to safely loop through any dataset type and dynamically create index patterns.  Otherwise if I revert it back to a string, it would still be "incorrect" and I'd need to update the code.

Reply:
I'll move that type back to string and then create a type for the index pattern names I plan to create.  I think that is probably what I'm looking for.

Reply:
Ok, I'll leave remove DatasetType and perhaps add an index pattern type to keep track of the index pattern names that will be created.  Can update later if I should just be creating without checking.  Thanks!